sda-commons-server-auth-testing/src/main/java/org/sdase/commons/server/opa/testing/OpaRule.java

```diff
@@ -75,18 +86,59 @@ protected void after() {
     wire.stop();
   }
 
-  public void verify(int count, String httpMethod, String paths) {
-    wire.verify(count, matchForVerify(httpMethod, splitPath(paths)));
+  /**
+   * Verify if the policy was called for the method/path
+   *
+   * <p>Please prefer to use {@code verify(count, onRequest().withMethod(method).withPath(path))}
+   * instead.
+   *
+   * @param count the expected count of calls
+   * @param httpMethod the HTTP method to check
+   * @param path the path to check
+   */
+  public void verify(int count, String httpMethod, String path) {
+    verify(count, onRequest().withMethod(httpMethod).withPath(path));
+  }
+
+  public void verify(int count, AllowBuilder allowBuilder) {
+    verify(count, (StubBuilder) allowBuilder);
+  }
+
+  private void verify(int count, StubBuilder builder) {
+    RequestPatternBuilder requestPattern;
+
+    if (builder.onAnyRequest) {
+      requestPattern =
+          RequestPatternBuilder.newRequestPattern(RequestMethod.POST, urlMatching("/.*"));
+    } else {
+      requestPattern =
+          RequestPatternBuilder.newRequestPattern(RequestMethod.POST, urlMatching("/.*"))
+              .withRequestBody(matchingJsonPath("$.input.httpMethod", equalTo(builder.httpMethod)))
+              .withRequestBody(
+                  matchingJsonPath("$.input.path", equalToJson(getJson(builder.paths))));
+
+      if (builder.matchJWT) {
+        requestPattern.withRequestBody(matchingJsonPath("$.input.jwt", equalTo(builder.jwt)));
+      }
+    }
+
+    wire.verify(count, requestPattern);
   }
 
   public void mock(BuildBuilder builder) {
     builder.build(wire);
   }
 
-  private RequestPatternBuilder matchForVerify(String httpMethod, String[] paths) {
-    return RequestPatternBuilder.newRequestPattern(RequestMethod.POST, urlMatching("/.*"))
-        .withRequestBody(matchingJsonPath("$.input.httpMethod", equalTo(httpMethod)))
-        .withRequestBody(matchingJsonPath("$.input.path", equalToJson(getJson(paths))));
+  public interface RequestMethodBuilder {
+    RequestPathBuilder withMethod(String method);
+  }
+
+  public interface RequestPathBuilder {
+    RequestExtraBuilder withPath(String path);
+  }
+
+  public interface RequestExtraBuilder extends AllowBuilder {
+    RequestExtraBuilder withJwt(String jwt);
```

Comment:
Why does it return the `RequestExtraBuilder`? This seems to allow multiple calls to `withJwt(…)` in the builder. Does that makes sense?

Reply:
No, that would not make sense. In future, we could also add the possibility to check for e.g. request headers. We could then extend this interface.